src/main/java/org/folio/event/handler/HoldingUpdateAsyncRecordHandler.java

```diff
@@ -0,0 +1,193 @@
+package org.folio.event.handler;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.impl.headers.HeadersMultiMap;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.WebClient;
+import lombok.extern.log4j.Log4j2;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.folio.cql2pgjson.exception.FieldException;
+import org.folio.event.dto.ResourceEvent;
+import org.folio.event.service.AuditOutboxService;
+import org.folio.okapi.common.GenericCompositeFuture;
+import org.folio.okapi.common.XOkapiHeaders;
+import org.folio.rest.jaxrs.model.Location;
+import org.folio.rest.jaxrs.model.OrderLineAuditEvent;
+import org.folio.rest.jaxrs.model.PoLine;
+import org.folio.rest.persist.Conn;
+import org.folio.rest.persist.DBClient;
+import org.folio.services.lines.PoLinesService;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.folio.event.InventoryEventType.INVENTORY_HOLDING_UPDATE;
+
+@Log4j2
+public class HoldingUpdateAsyncRecordHandler extends InventoryUpdateAsyncRecordHandler {
+
+  public static final String ID = "id";
+  public static final String INSTANCE_ID = "instanceId";
+  public static final String PERMANENT_LOCATION_ID = "permanentLocationId";
+  public static final String PO_LINE_LOCATIONS_HOLDING_ID_CQL = "locations==*%s*";
+  public static final String STORAGE_HOLDING_URL = "%s/holdings-storage/holdings/%s";
```

Comment:
and using bulk endpoint to update holdings was not possible?

Reply:
> and using bulk endpoint to update holdings was not possible?

I will try it again